scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -1249,65 +1385,110 @@ public DtlsConnectorConfig build() {
 			if (config.staleConnectionThreshold == null) {
 				config.staleConnectionThreshold = DEFAULT_STALE_CONNECTION_TRESHOLD;
 			}
-			if (config.certificateVerifier == null && config.trustStore != null) {
-				config.certificateVerifier = new StaticCertificateVerifier(config.trustStore);
-			}
 			if (config.sniEnabled == null) {
 				config.sniEnabled = Boolean.TRUE;
 			}
 			if (config.verifyPeersOnResumptionThreshold == null) {
 				config.verifyPeersOnResumptionThreshold = DEFAULT_VERIFY_PEERS_ON_RESUMPTION_THRESHOLD_IN_PERCENT;
 			}
+			if (config.certificateVerifier == null && config.trustStore != null) {
+				config.certificateVerifier = new StaticCertificateVerifier(config.trustStore);
+			}
+			if (config.trustCertificateTypes == null) {
+				if (config.trustedRPKs != null) {
+					config.trustCertificateTypes = ArrayUtils.add(config.trustCertificateTypes,
+							CertificateType.RAW_PUBLIC_KEY);
+				}
+				if (config.certificateVerifier != null) {
+					config.trustCertificateTypes = ArrayUtils.add(config.trustCertificateTypes, CertificateType.X_509);
+				}
+			} else {
+				if (ArrayUtils.contains(config.trustCertificateTypes, CertificateType.RAW_PUBLIC_KEY)) {
+					if (config.trustedRPKs == null) {
```

Comment:
why not trusting All RPK by default like before ?

Reply:
My first idea was to propose a builder with maximum default value (and so minimal needed configuration) to be able to have a testable/running application quickly.

But In that case, maybe it's better to force user to ask himself some question about what it trust or not. :thinking:

Reply:
The behaviour before is also more based on the history than a common way.
Historical, all RPK's are trusted, but X.509 need special trust. Then it ge's changed into, RPK may be trusted and x.509 may be trusted in general. And both with different defaults (RPK: trust all, x.509: trust none). So I would prefer to change the behaviour to a common model.

Reply:
Agree for consistency.
Agree for forcing users to explicitly say what they want to trust.